core/src/cpu/air/mod.rs

```diff
@@ -420,6 +434,53 @@ impl CpuChip {
             .assert_eq(local.pc, next.pc);
     }
 
+    /// Constraints related to the commit instruction.
+    pub(crate) fn commit_eval<AB: SP1AirBuilder>(
+        &self,
+        builder: &mut AB,
+        local: &CpuCols<AB::Var>,
+        is_commit: AB::Expr,
+    ) {
+        let public_values = builder.public_values();
+
+        // Convert public values into words.
+        let mut digest_words = Vec::new();
```

Comment:
It seems like this type of logic won't work when we start adding new public inputs. Do we need a higher level abstraction for PublicInputs?

Reply:
Maybe we should have PublicValuesCols with transmute so that everything is labeled.

Reply:
I feel like this should be done in the follow up PRs to this one that adds other public inputs.  This PR is already very big.

Reply:
Sounds good.